src/molecules/Comp2Param.cpp

```diff
@@ -40,13 +41,23 @@ void Comp2Param::initialize(
 		for (unsigned int compj = compi + 1; compj < m_numcomp; ++compj) {
 			ParaStrm& pstrmij = m_ssparatbl(compi, compj);
 			unsigned int ncj = components[compj].numLJcenters();
-			double xi = *mixpos;
-			++mixpos;
-			double eta = *mixpos;
-			++mixpos;
+			const auto mixingrule = mixcoeff[compi][compj];
+			// Generic mixing functions
+			std::function<double(double, double)> mixingSigma;
+			std::function<double(double, double)> mixingEpsilon;
+			// Get parameters
+			if (mixingrule->getType() == "LB") {
+				const double eta = mixingrule->getParameters().at(0);
+				const double xi  = mixingrule->getParameters().at(1);
+				mixingEpsilon = [=](double epsi, double epsj) { return xi * sqrt(epsi * epsj); };
+				mixingSigma = [=](double sigi, double sigj) { return eta * (sigi + sigj); };
 #ifndef NDEBUG
-			Log::global_log->info() << "cid+1(compi)=" << compi+1 << " <--> cid+1(compj)=" << compj+1 << ": xi=" << xi << ", eta=" << eta << std::endl;
+				Log::global_log->info() << "Mixing : cid+1(compi)=" << compi+1 << " <--> cid+1(compj)=" << compj+1 << ": xi=" << xi << ", eta=" << eta << std::endl;
 #endif
+			} else {
+				Log::global_log->error() << "Mixing: Only LB rule supported" << std::endl;
+				Simulation::exit(1);
+			}
```

Comment:
you can be even more fancy:

```suggestion
const auto [mixingSigma, mixingEpsilon] =
				[&]() -> std::pair<std::function<double(double, double)>, std::function<double(double, double)>> {
				// Get parameters
				if (mixingrule->getType() == "LB") {
					const double eta = mixingrule->getParameters().at(0);
					const double xi = mixingrule->getParameters().at(1);
#ifndef NDEBUG
					Log::global_log->info()
						<< "Mixing : cid+1(compi)=" << compi + 1 << " <--> cid+1(compj)=" << compj + 1 << ": xi=" << xi
						<< ", eta=" << eta << std::endl;
#endif
					return {[=](double epsi, double epsj) { return xi * sqrt(epsi * epsj); },
							[=](double sigi, double sigj) { return eta * (sigi + sigj); }};
				} else {
					Log::global_log->error() << "Mixing: Only LB rule supported" << std::endl;
					Simulation::exit(1);
					return {};
				}
			}();
```

This has the following advantages:
- the functions are (in the scope of the for loop) are directly defined where they are declared
- clearer scoping
- they are `const`